src/classiceditorui.js

```diff
@@ -89,6 +104,35 @@ export default class ClassicEditorUI {
 				}
 
 				return Promise.all( promises );
+			} )
+			.then( () => {
+				const toolbarFocusTracker = this.view.toolbar.focusTracker;
+
+				// Because toolbar items can get focus, the overall state of
+				// the toolbar must also be tracked.
+				this.focusTracker.add( this.view.toolbar.element );
+
+				// Listen on the keystrokes from the main UI.
+				this.keystrokes.listenTo( this.view.element );
+
+				// Listen on the keystrokes from the floating panels, toolbars and the such.
+				this.keystrokes.listenTo( this.view._bodyCollectionContainer );
+
+				// Focus the toolbar on the keystroke, if not already focused.
+				this.keystrokes.set( 'alt + f10', ( data, cancel ) => {
+					if ( this.focusTracker.isFocused && !toolbarFocusTracker.isFocused ) {
+						this.view.toolbar.focus();
+						cancel();
+					}
+				} );
+
+				// Blur the toolbar and bring the focus back to editable on the keystroke.
+				this.keystrokes.set( 'esc', ( data, cancel ) => {
```

Comment:
If you'll agree with me that editor UI doesn't need its own keystroke handler, then you'll have to do something with this listener. I even worried a bit about that, but `ToolbarView` has its own handler so we can listen to <kbd>Esc</kbd> using that one.